applications/StructuralMechanicsApplication/tests/cpp_tests/test_total_lagrangian.cpp

```diff
@@ -0,0 +1,116 @@
+
+        for (auto& r_node : r_model_part.Nodes()){
+            r_node.AddDof(DISPLACEMENT_X);
+            r_node.AddDof(DISPLACEMENT_Y);
+            r_node.AddDof(DISPLACEMENT_Z);
+        }
+
+        std::vector<ModelPart::IndexType> element_nodes {1,2,3};
+        auto p_element = r_model_part.CreateNewElement("TotalLagrangianElement2D3N", 1, element_nodes, p_elem_prop);
+
+        p_element->Initialize(r_model_part.GetProcessInfo());
+
+        //define a matrix A and impose that the displacement on each node is u = A*x0
+        Matrix A0 = ZeroMatrix(3,3);
+        A0(0,0) = 2e-2;
+        A0(0,1) = 5e-2;
+        A0(1,0) = 5e-2;
+        A0(1,1) = -1e-2;
+        A0(2,2) = 1.0;
+
+        Matrix lhs;
+        Vector rhs;
+        const auto& const_procinfo_ref = r_model_part.GetProcessInfo();
+        double factor = 1.0;
+        for(unsigned int step=0; step<5; step++)
+        {
+            factor+=1.0;
+            Matrix A = factor*A0;
+            for (auto& r_node : r_model_part.Nodes()){
+                noalias(r_node.GetSolutionStepValue(DISPLACEMENT)) = prod(A, r_node.GetInitialPosition());
+                r_node.Coordinates() = r_node.GetInitialPosition() + r_node.GetSolutionStepValue(DISPLACEMENT); //here i update the node coordinates
+            }
+
+            p_element->InitializeSolutionStep(const_procinfo_ref);
+            p_element->InitializeNonLinearIteration(const_procinfo_ref);
+            p_element->CalculateLocalSystem(lhs,rhs,const_procinfo_ref);
+            p_element->FinalizeNonLinearIteration(const_procinfo_ref);
+            p_element->FinalizeSolutionStep(const_procinfo_ref);
+
+            //known A we can compute the deformation gradient analytically as F = A + I and from that the analytical value of the strain
+            Matrix F = IdentityMatrix(3,3) + A;
+            Matrix strain_mat = 0.5*(prod(trans(F),F)-IdentityMatrix(3,3));
+            std::vector<double> reference_strain = {strain_mat(0,0), strain_mat(1,1), 2.0*strain_mat(0,1)};
+
+            //also the stress can be computed analytically as
+            Vector reference_stress(3);
+            reference_stress[0] = c1 * reference_strain[0] + c2 * reference_strain[1];
+            reference_stress[1] = c2 * reference_strain[0] + c1 * reference_strain[1];
+            reference_stress[2] = c3 * reference_strain[2];
+
+            std::vector<Vector> output_strains(1);
+            p_element->CalculateOnIntegrationPoints(GREEN_LAGRANGE_STRAIN_VECTOR,output_strains, r_model_part.GetProcessInfo());
+
+            std::vector<Vector> output_stress(1);
+            p_element->CalculateOnIntegrationPoints(PK2_STRESS_VECTOR,output_stress, r_model_part.GetProcessInfo());
+            
+            for(unsigned int i=0; i<reference_strain.size(); ++i)
+            {
+                KRATOS_CHECK_DOUBLE_EQUAL(output_strains[0][i], reference_strain[i]);
+                KRATOS_CHECK_DOUBLE_EQUAL(output_stress[0][i], reference_stress[i]);
```

Comment:
I think here you can use the [vector check macro](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/checks.h#L58-L103)